release-tool.ps1

```diff
@@ -0,0 +1,456 @@
+    # Bootstrap the specified VS Toolchain
+    Import-Module "$VSBaseDir\Common7\Tools\Microsoft.VisualStudio.DevShell.dll"
+    Enter-VsDevShell -VsInstallPath $VSBaseDir -Arch $Arch -StartInPath $Path | Write-Host
+    Write-Host # Newline after command output
+}
+
+function Invoke-Cmd([string] $command, [string[]] $options = @(), [switch] $maskargs, [switch] $quiet) {
+    $call = ('{0} {1}' -f $command, ($options -Join ' '))
+    if ($maskargs) {
+        Write-Host "$command <masked>" -ForegroundColor DarkGray
+    }
+    else {
+        Write-Host $call -ForegroundColor DarkGray
+    }
+    if ($quiet) {
+        Invoke-Expression $call > $null
+    } else {
+        Invoke-Expression $call
+    }
+    if ($LASTEXITCODE -ne 0) {
+        throw "Failed to run command: {0}" -f $command
+    }
+    Write-Host #insert newline after command output
+}
+
+function Invoke-SignFiles([string[]] $files, [string] $key, [string] $time) {
+    if (-not (Test-Path -Path "$key" -PathType leaf)) {
+        throw "Appsign key file was not found! ($key)"
+    }
+    if ($files.Length -eq 0) {
+        return
+    }
+
+    Write-Host "Signing files using $key"  -ForegroundColor Cyan
+    $KeyPassword = Read-Host "Key password: " -MaskInput
+
+    foreach ($_ in $files) {
+        Write-Host "Signing file '$_' using Microsoft signtool..."
+        Invoke-Cmd "signtool" "sign -f `"$key`" -p `"$KeyPassword`" -d `"KeePassXC`" -td sha256 -fd sha256 -tr `"$time`" `"$_`"" -maskargs
+    }
+}
+
+function Invoke-GpgSignFiles([string[]] $files, [string] $key) {
+    if ($files.Length -eq 0) {
+        return
+    }
+
+    Write-Host "Signing files using GPG key $key" -ForegroundColor Cyan
+
+    foreach ($_ in $files) {
+        Write-Host "Signing file '$_' and creating DIGEST..."
+        Remove-Item "$_.sig"
+        Invoke-Cmd "gpg" "--output `"$_.sig`" --armor --local-user `"$key`" --detach-sig `"$_`""
+        $FileName = (Get-Item $_).Name
+        (Get-FileHash "$_" SHA256).Hash + " *$FileName" | Out-File "$_.DIGEST" -NoNewline
+    }
+}
+
+
+# Handle errors and restore state
+$OrigDir = $(Get-Location).Path
+$OrigBranch = $(git rev-parse --abbrev-ref HEAD)
+$ErrorActionPreference = 'Stop'
+trap {
+    Write-Host "Restoring state..." -ForegroundColor Yellow
+    $(git checkout $OrigBranch)
+    Set-Location "$OrigDir"
+}
+
+Write-Host "KeePassXC Release Preparation Helper" -ForegroundColor Green
+Write-Host "Copyright (C) 2021 KeePassXC Team <https://keepassxc.org/>`n" -ForegroundColor Green
+
+# Prepend extra PATH locations as specified
+if ($ExtraPath) {
+    $env:Path = "$ExtraPath;$env:Path"
+}
+
+# Resolve absolute directory for paths
+$SourceDir = (Resolve-Path $SourceDir).Path
+
+# Check format of -Version
+if ($Version -notmatch "^\d+\.\d+\.\d+$") {
+    throw "Invalid format for -Version input"
+}
+
+# Check platform
+if (-not $IsWindows) {
+    throw "The powershell release tool is not available for Linux or macOS at this time."
+}
+
+if ($Merge) {
```

Comment:
Can you add ASCII art comment blocks between commands, so they can be more easily separated (like we did in the original release-tool)?

Reply:
Not sure what you mean, add [INFO] in front of general messages? I have general messages in Aqua color and command strings in dark gray. Command output is in White. It makes it easy to read what's going on.

![image](https://user-images.githubusercontent.com/2809491/151617696-29df588a-3a9a-4284-b9bf-9ec5313b876d.png)

Reply:
I mean in the code, just so we can visually distinguish the beginnings and ends of a command in the script.

Reply:
I mean in the code, just so we can visually distinguish the beginnings and ends of a command in the script.